client/components/data-labels/labels.directive.js

```diff
@@ -90,6 +101,57 @@ angular.module('probedock.dataLabels').directive('simpleLabel', function() {
       return str;
     }
   };
+})
+.controller('SourceUrlLabelCtrl', function($scope) {
+  _.defaults($scope, {
+    tooltipPlacement: 'top'
+  });
+
+  $scope.isDirty = function() {
+    // Check if SCM data are present
+    if ($scope.scm) {
+      // Check if dirty is set
+      if (!_.isUndefined($scope.scm.dirty)) {
+        return $scope.scm.dirty;
+      } else if (!_.isUndefined($scope.scm.remote)) { // Check if remote data are present to consider the commit as dirty or not
+        if (!_.isUndefined($scope.scm.remote.behind)) { // Check if commit is behind remote
+          return $scope.scm.remote.behind > 0;
```

Comment:
If one is behind but not dirty, it is guaranteed that the commit is on the remote, therefore I would not return true in this case.